lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -231,26 +237,28 @@ std::optional<DecompressResult> ImageDecoderImpeller::DecompressTexture(
 
   auto buffer = scaled_allocator->GetDeviceBuffer();
   if (!buffer.has_value()) {
-    return std::nullopt;
+    return DecompressResult{.decode_error = "Unable to get device buffer"};
   }
   return DecompressResult{.device_buffer = buffer.value(),
                           .sk_bitmap = scaled_bitmap,
                           .image_info = scaled_bitmap->info()};
 }
 
-sk_sp<DlImage> ImageDecoderImpeller::UploadTextureToPrivate(
+std::pair<sk_sp<DlImage>, std::string>
+ImageDecoderImpeller::UploadTextureToPrivate(
     const std::shared_ptr<impeller::Context>& context,
     const std::shared_ptr<impeller::DeviceBuffer>& buffer,
     const SkImageInfo& image_info) {
   TRACE_EVENT0("impeller", __FUNCTION__);
   if (!context || !buffer) {
-    return nullptr;
+    return std::make_pair(nullptr, "No Impeller context or buffer available");
```

Comment:
It would be helpful to split these into separate messages - for example if there is no context there are differnet causes then if the buffer is invalid.